src/functions/metadata.ts

```diff
@@ -153,7 +153,11 @@ export const getOnchainMetadata = (
         onchainMetadataResult = foundMetadata;
         isFound = true;
       } else {
-        // fallback
+        // Fallback
+        // Due to previous incorrect implementation, metadata passed as version 2 needed to have
+        // text-encoded policies/asset names (instead of correct byte-encoded).
+        // To prevent breaking change, in case we don't find metadata for an asset under
+        // correct byte-encoded key, we fallback to version 1 text-encoded format.
```

Comment:
```suggestion
        // Due to previous incorrect/incomplete implementation, metadata with
        // text-encoded policies/asset names (instead of correct byte-encoded)
        // incorrectly passed the verification since CIP25v2,i.e. "version": 2 was not yet implemented.
        // To prevent breaking change, in case we don't find metadata for an asset under
        // correct byte-encoded key, we fallback to version 1 text-encoded format.
```